src/sources/internal_logs.rs

```diff
@@ -64,26 +65,26 @@ async fn run(
         .chain(tokio_stream::wrappers::BroadcastStream::new(
             subscription.receiver,
         ))
+        .filter_map(|log| future::ready(log.ok()))
+        .filter(|log| future::ready(!log.contains("internal")))
         .take_until(shutdown);
 
     // Note: This loop, or anything called within it, MUST NOT generate
     // any logs that don't break the loop, as that could cause an
     // infinite loop since it receives all such logs.
```

Comment:
Funny situation. I think the compromise here of still logging but filtering `internal_logs` own logs from its own processing makes sense. Maybe it could cause some minor confusion down the line when using the `internal_logs` source if users then don't see all the logs they'd see in stdout? Sorry, not an actionable comment, just some random thoughts.

Reply:
Well, I'm a bit concerned with this `internal_logs` source and every component that comes after...
Let say we have `internal_logs` => `transform` => `sink`.
Each log received will be sent to the transform, that will emit an event, this event will be sent to `internal_logs`, triggering a log in the transform and so on. Same for the sink...

Reply:
Exactly why this path _must not_ emit any logs itself. As long as that remains true, this infinite loop shouldn't happen.

Reply:
by `path` do you mean even the `transform` and `sink`? If so, do we have a way to block that?